From c13a3842c31dc76a6900aa4ad86e0159c876d811 Mon Sep 17 00:00:00 2001
From: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
Date: Mon, 3 Jul 2023 16:09:29 +0530
Subject: [PATCH 08/16] wifi: mac80211: fix AddBA response addressing

Since this frame is addressed from/to an MLD, it should be
built with the correct AP MLD address (in station mode) to
be encrypted properly.

Signed-off-by: Johannes Berg <johannes.berg@intel.com>
Signed-off-by: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
---
 net/mac80211/agg-rx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/agg-rx.c b/net/mac80211/agg-rx.c
index b4ed3ce..31ab96e 100644
--- a/net/mac80211/agg-rx.c
+++ b/net/mac80211/agg-rx.c
@@ -239,7 +239,7 @@ static void ieee80211_send_addba_resp(struct sta_info *sta, u8 *da, u16 tid,
 		sdata->vif.type == NL80211_IFTYPE_MESH_POINT) {
 		memcpy(mgmt->bssid, sdata->vif.addr, ETH_ALEN);
 	} else if (sdata->vif.type == NL80211_IFTYPE_STATION) {
-		memcpy(mgmt->bssid, sdata->deflink.u.mgd.bssid, ETH_ALEN);
+		memcpy(mgmt->bssid, sdata->vif.cfg.ap_addr, ETH_ALEN);
 	} else if (sdata->vif.type == NL80211_IFTYPE_ADHOC) {
 		memcpy(mgmt->bssid, sdata->u.ibss.bssid, ETH_ALEN);
 	}
-- 
2.17.1

